ledger-service/http-json-testing/src/main/scala/com/daml/http/HttpServiceTestFixture.scala

```diff
@@ -316,37 +316,39 @@ object HttpServiceTestFixture extends LazyLogging with Assertions with Inside {
   def jwtForParties(
       actAs: List[String],
       readAs: List[String],
-      ledgerId: String,
+      ledgerId: Option[String] = None,
       withoutNamespace: Boolean = false,
       admin: Boolean = false,
   ): Jwt = {
     import AuthServiceJWTCodec.JsonImplicits._
-    val payload =
-      if (withoutNamespace)
-        s"""{
-               |  "ledgerId": "$ledgerId",
-               |  "applicationId": "test",
-               |  "exp": 0,
-               |  "admin": $admin,
-               |  "actAs": ${actAs.toJson.prettyPrint},
-               |  "readAs": ${readAs.toJson.prettyPrint}
-               |}
-              """.stripMargin
-      else
-        (CustomDamlJWTPayload(
-          ledgerId = Some(ledgerId),
+    val payload: JsValue = {
+      val customJwtPayload: AuthServiceJWTPayload =
+        CustomDamlJWTPayload(
+          ledgerId = ledgerId,
           applicationId = Some("test"),
           actAs = actAs,
           participantId = None,
           exp = None,
-          admin = false,
+          admin = admin,
           readAs = readAs,
-        ): AuthServiceJWTPayload).toJson.prettyPrint
+        )
+      val payloadJson = customJwtPayload.toJson
+      if (withoutNamespace) {
+        // unsafe code but if someone changes the underlying structure
+        // they will notice the failing tests.
+        val payloadObj = payloadJson.asInstanceOf[JsObject]
```

Comment:
I'm missing why you first turn this into Json and then get stuff out of the Json in this way?

Reply:
Would it not make sense to add a method to (I guess) CustomDamlJWTPayload to 'remove' the  namespace, or have a version without the namespace?

Reply:
The idea was born out of a [comment I made](https://github.com/digital-asset/daml/pull/13161#discussion_r819583447), although I would have preferred a separate `JsonWriter` rather then stripping stuff from the `JsObject`. I don't think we should have a separate case for that as it's a now abandoned format and I'd rather have just the rendering for it, possibly restricted to the test scope.

Reply:
Because it is only for the test code & deprecated this should be fine for now. I don't want to write a separate json writer in the json api for the CustomClaimJwt because the idea here was to rely as much as possible on the existing and pressumably always right code from the ledger api.

Reply:
Was just curious, since this is testing code I'm not too worried, though you could have defined the JsonWriter only in the test.